server/map_rpc_server.go

```diff
@@ -166,6 +166,12 @@ func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapL
 			return nil, status.Errorf(codes.InvalidArgument,
 				"len(%x): %v, want %v", l.Index, got, want)
 		}
+		if l.LeafValue == nil {
```

Comment:
Storing empty leaves in maps is valid, we just don't want to put anything on disk for them.